src/core/shape_inference/include/shuffle_channels_shape_inference.hpp

```diff
@@ -0,0 +1,45 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <openvino/op/shuffle_channels.hpp>
+
+namespace ov {
+namespace op {
+namespace v0 {
+
+template <class T>
+void shape_infer(const ShuffleChannels* op, const std::vector<T>& input_shapes, std::vector<T>& output_shapes) {
+    NODE_VALIDATION_CHECK(op, input_shapes.size() == 1 && output_shapes.size() == 1);
+
+    const auto& group = op->get_group();
+    NODE_VALIDATION_CHECK(op, group >= 1, "The 'group' parameter must be greater or equal to 1.");
+
+    const auto& input_shape = input_shapes[0];
+    const auto input_shape_rank = input_shape.rank();
+
+    if (input_shape_rank.is_static()) {
+        const auto input_rank_value = input_shape.size();
+        NODE_VALIDATION_CHECK(op, input_rank_value >= 1, "The input tensor's shape is expected to be at least 1D.");
+
+        const auto& axis = op->get_axis();
+        size_t axis_zb = static_cast<size_t>(axis >= 0 ? axis : (axis + static_cast<int64_t>(input_rank_value)));
+        NODE_VALIDATION_CHECK(op,
+                              axis_zb < input_rank_value,
+                              "The 'axis' parameter for ShuffleChannels has to point to one of the "
+                              "input tensor's shape dimensions.");
```

Comment:
insufficient check. it assumes that `axis` is already in acceptable range
how about `axis = -8` and shape `[1, 2, 3]`?